clang/lib/Basic/Targets/Mips.h

```diff
@@ -196,7 +196,7 @@ class LLVM_LIBRARY_VISIBILITY MipsTargetInfo : public TargetInfo {
       ABI = Name;
       return true;
     }
-    if (Name == "purecap") {
+    if (Name == "purecap" || Name == "cherios") {
```

Comment:
Glancing at this diff, the only difference I see between purecap and cherios is the PIC hack that should be fixed? Either way, in my opinion, putting this in `-mabi=` is wrong, and you should be keying off the OS in the target triple instead. Similarly, why do we need separate a CHERIOSABI ELF flag rather than putting CheriOS in the OS field?

Reply:
The other place I look for the ABI being cherios is MipsABIInfo::computeTargetABI, which creates a MipsABIInfo object which is used everywhere to do CheriOS specific things. The target triple is available there so could also be keying off the OS field. I wonder if making this change would not mean that there are lots of places that will read OS is CheriOS || OS is FreeBSD instead, in which one is neater. Really it is a change to both the ABI and OS.